### PR TITLE
Lint SQL files

### PR DESCRIPTION
**Pull request by SimonRichardson**
~~Requires https://github.com/juju/juju/pull/17428 to land first~~

----

This formats and lints all the SQL files. We're using sqlfluff for
linting and formatting. The style isn't 100% nice, but then again
any formatter isn't perfect (see gofmt as an example).

The files are 100% consistent though.

In addition to formatting, I found a lot of stupid and simple mistakes
that the linter found.

 - using keywords for column names
 - missing commas
 - ambiguous column names in joins
 - duplicated columns in views

I'm not even sure how some of the foreign keys worked (unless the sqlite
parser is very lenient) because missing commas were present in multiple
locations.

The point of this is to improve the correctness of our SQL
statements.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

Install sqlfluff.

Use an env rather than stomping on the system is probably wise.

```sh
$ pip install sqlfluff
```

## Links

**Jira card:** JUJU-

### CODE REVIEW

File: domain/schema/controller/sql/0002-lease.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    uuid TEXT PRIMARY KEY,
+    lease_type_id INT NOT NULL,
+    model_uuid TEXT,
+    name TEXT,
+    holder TEXT,
+    start TIMESTAMP,
+    expiry TIMESTAMP,
+    CONSTRAINT fk_lease_lease_type
+    FOREIGN KEY (lease_type_id)
+    REFERENCES lease_type (id)
```
Comment (nvinuesa) [inline domain/schema/controller/sql/0002-lease.sql:23]: This format :nauseated_face:

File: .github/.sqlfluff [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[sqlfluff]
+dialect = sqlite
+max_line_length = 180
```
Comment (nvinuesa) [inline .github/.sqlfluff:3]: Can we get this down to 80, or at most 120?

**Review by nvinuesa — APPROVED** [review]
Indeed, formatting doesn't look good but it's a good tradeoff for consistency. 
I'd get the `max_line_length` to 80 or 120.

File: .github/.sqlfluff (hunk 2 above)
Comment (SimonRichardson) [reply .github/.sqlfluff:3]: I've left it at this because the changelog namespaces get rendered in the most ugliest way possible. I have my limits!

**Comment by SimonRichardson** [comment]
> Do they have a way to change the rendering already?

I've changed the rendering with the config, but I can't change it to what we had. There are issues tracking what we want, just not implemented yet.

 - https://github.com/sqlfluff/sqlfluff/issues/5855
 - https://github.com/sqlfluff/sqlfluff/issues/5609

**Pull request merged**